### PR TITLE
changes in RoomInfo in KssuFile.h Tile8bpp in nitro.h

### PR DESCRIPTION
**Pull request by Rinkaa**
1.Change RoomInfo from union to struct, with small fixes in detail.
2.Tile8bpp do not rely on KfPlt , a new TemplateParam has been set to handle the type of plt.
3.remove"winres.rc" from MApRX_qt.pro

### CODE REVIEW

File: src/KssuFile.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include "NitroLz.h"
 #include <assert.h>
 #include <tuple>
+#include <cstring>
```
Comment (wwylele) [inline src/KssuFile.cpp:26]: Err where is this used?

File: src/KssuFile.cpp [hunk 2]
```diff
@@ -492,8 +493,8 @@ void KfMap::newItem(u8 before_itemId,const RipeItem& item){
     metaData.itemCount++;
 }
 
-void KfMap::resizeMap(u16 width, u16 height,
-               int x0, int y0//Where to put old cell(0,0) on new map
+void KfMap::resizeMap(u8 width,u8 height,
+               int x0,int y0//Where to put old cell(0,0) on new map
```
Comment (wwylele) [inline src/KssuFile.cpp]: width and height should be u16. also the code below

File: src/KssuFile.cpp [hunk 3]
```diff
@@ -513,14 +514,10 @@ void KfMap::resizeMap(u16 width, u16 height,
         item.basic.x+=x0*24;
         item.basic.y+=y0*24;
     }
-    forEachCellReferenceInScripts([x0,y0](u16& x,u16 &y){
-        x+=x0;
-        y+=y0;
-    });
```
Comment (wwylele) [inline src/KssuFile.cpp:519]: I need this

File: src/KssuFile.h [hunk 4]
```diff
@@ -262,33 +262,11 @@ class KfMap{
...
-            }
-        };
-        for(RipeCell & cell:cells){
-            f(cell.scripts);
-        }
-        for(RipeItem & item:items){
-            f(item.scripts);
-        }
-    }
-
```
Comment (wwylele) [inline src/KssuFile.h:286]: I need this

File: src/KssuFile.cpp (hunk 1 above)
Comment (Rinkaa) [inline src/KssuFile.cpp:26]: memcpy etc

**Pull request merged**